stacks-signer/src/signerdb.rs

```diff
@@ -83,31 +72,6 @@ impl SignerDb {
         )
     }
 
-    /// Get the signer state for the provided reward cycle if it exists in the database
-    pub fn get_signer_state(&self, reward_cycle: u64) -> Result<Option<SignerState>, DBError> {
```

Comment:
I think it's fine to keep a variation of this -- you'd just store your encrypted state to the DB instead of the cleartext state.  However, it should be something performed _in addition to_ saving to StackerDB.

Reply:
To be clear, I stand by my initial claim that storing locally is redundant if you're storing to StackerDB.  However, a lack of signer state availability is such a catastrophic failure condition that keeping a copy around locally isn't the worst idea, especially for operators who run their Stacks nodes in different hosts (or even different networks), which can go offline.  Also, if there's a network-wide Stacks node crash, then keeping the signer state locally can at least provide one additional avenue for the signers to boot back up.

Reply:
Updated. The logic is now:

Persistence:
1. Attempt to persist in both StackerDB and in the local sqlite database.
2. Return both results to the caller, which will now panic if either operation fails (this was panicking failure before as well, but I'd be open to explore if we'd like the signer to continue if one of the storage mechanisms work)

Loading:
Attempt to load the state from stackerdb first, write a warning if this fails and secondly attempt to load the state from the local database. Panic if both fails.

Let me know if you want the logic to be different or have any thoughts on how it is implemented.